Source/ui_unix/mainwindow.h

```diff
@@ -66,12 +66,16 @@ class MainWindow : public QMainWindow
     struct lastOpenCommand
     {
         BootType type;
-        const char* filename;
-
-        lastOpenCommand(BootType m_type, const char* m_filename) : type(m_type),filename(m_filename){}
+		std::string filename;
+		lastOpenCommand() = default;
+		void reset(BootType m_type, std::string m_filename)
```

Comment:
Why have a reset function instead of a constructor that takes 2 parameters? Also, I would expect a 'reset' function to clear all members/release all resources/etc..

Reply:
fail on my part, not sure why but I initially tried using a constructor and it failed, so I went straight to this, but it's working now, ill push fix in few.